### PR TITLE
014 - Kakuna - Sessions & auth security

### PR DESCRIPTION
**Pull request by zopa-simon-cass**
Unsure if we've properly met the req for redirects, this temporary holder will do for now

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
   helper_method :current_user
 
   def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
+
+  def authorize
+    redirect_to login_path, alert: 'You must login to access this page.' if
+    current_user.nil?
+  end
```
Comment (Sindex42) [inline app/controllers/application_controller.rb]: Can we get rid of authorize?

File: app/helpers/posts_helper.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (Sindex42) [inline app/helpers/posts_helper.rb]: Do we need this file? Or should we move the helper method form the posts controller here?

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
```
Comment (Sindex42) [inline app/helpers/sessions_helper.rb:2]: Same as above, should we get rid of this file?

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
 <% @posts.each do |post| %>
   <p><%= post.message %></p>
 <% end %>
 
 <%= link_to new_post_path do %>
   New post
 <% end %>
+<% else %>
+You must be signed in to view these posts
+<% end %>
```
Comment (Sindex42) [inline app/views/posts/index.html.erb]: Do we still need these bits as it's covered by the before_action in the controller?

File: config/routes.rb [hunk 5]
```diff
@@ -1,12 +1,17 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  get 'pages/secret'
+
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (Sindex42) [inline config/routes.rb]: Is this route being used?

**Review by Sindex42 — CHANGES_REQUESTED** [review]

**Review by Sindex42 — APPROVED** [review]

**Comment by anhuiyang** [comment]
changes updated, all resolved:)

**Review by anhuiyang — APPROVED** [review]
happy

**Review by anhuiyang — APPROVED** [review]

**Pull request merged**